File: swapNodePairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode* p = head;
        int t;
        while(p && p->next){
            t = p->val;
            p->val = p->next->val;
            p->next->val = t;
            p = p->next->next;
        }
        return head;
    }
};

File: swapNodePairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        p = head
        while p and p.next:
            t = p.val
            p.val = p.next.val
            p.next.val = t
            p = p.next.next
        
        return head